### PR TITLE
🧪 Add tests for /api/search route

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** The /api/search GET endpoint was missing tests. This PR adds unit tests covering query parsing, error handling, and backend interactions.
📊 **Coverage:** Covered happy paths (valid `q` and `maxResults`), edge cases (bounds checking on `maxResults`, missing `q`), and error conditions (backend errors and unknown exceptions).
✨ **Result:** 100% test coverage for `packages/web/src/app/api/search/route.ts`.

---
*PR created automatically by Jules for task [17071583619169858139](https://jules.google.com/task/17071583619169858139) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

`/api/search` の GET エンドポイントに対する単体テストを追加しています。
- `q` と `maxResults` の解析および上限・下限処理を検証
- 検索成功時のレスポンスとバックエンド呼び出しを検証
- 必須パラメータ不足とバックエンド例外時のエラー処理を検証
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

このPRは安全にマージできると考えられます。

ブロッキングとなる不具合は残っていません。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/web/src/app/api/search/route.test.ts | `/api/search` の正常系、入力境界値、必須パラメータ不足、バックエンド障害を網羅するテストを追加しています。 |

</details>

<sub>Reviews (2): Last reviewed commit: ["test(web): align search route tests with..."](https://github.com/hiroki-org/paper-tools/commit/467a70737cbdfa4aeabfe58c511214ab089622ee) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=47324997)</sub>

**Context used:**

- Knowledge Base — [Web app (`packages/web`)](https://app.greptile.com/hiroki-org/-/custom-context/knowledge-base/hiroki-org/paper-tools/-/docs/web.md)

### CODE REVIEW

File: packages/web/src/app/api/search/route.test.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+describe("/api/search GET", () => {
+    beforeEach(() => {
+        vi.clearAllMocks();
+    });
+
+    it("qパラメータのみ指定でsearchByKeywordがデフォルトmaxResults(30)で呼ばれる", async () => {
+        const mockPapers = [{ paperId: "paper1", title: "Paper 1" }];
+        vi.mocked(searchByKeyword).mockResolvedValueOnce(mockPapers as any);
+
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/app/api/search/route.test.ts]: Fixed in 467a707: fixtures now use `Awaited<ReturnType<typeof searchByKeyword>>`, include the required `authors` field, and no longer use `as any`.

File: packages/web/src/app/api/search/route.test.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { NextRequest } from "next/server";
+
+vi.mock("@paper-tools/drilldown", () => ({
+    searchByKeyword: vi.fn(),
+}));
+
+const { searchByKeyword } = await import("@paper-tools/drilldown");
+const { GET } = await import("./route");
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/app/api/search/route.test.ts]: Fixed in 467a707: the relative dynamic import now uses `./route.js`.

[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/app/api/search/route.test.ts]: Fixed in 467a707: the route import now explicitly uses the `.js` extension.

File: packages/web/src/app/api/search/route.test.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import { beforeEach, describe, expect, it, vi } from "vitest";
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/app/api/search/route.test.ts]: Fixed in 467a707: the test imports only `vi` and relies on the configured Vitest globals.

**Comment by is0692vs** [comment]
@greptile review

File: packages/web/src/app/api/search/route.test.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+describe("/api/search GET", () => {
+    beforeEach(() => {
+        vi.clearAllMocks();
+    });
+
+    it("qパラメータのみ指定でsearchByKeywordがデフォルトmaxResults(30)で呼ばれる", async () => {
+        const mockPapers = [{ paperId: "paper1", title: "Paper 1" }];
+        vi.mocked(searchByKeyword).mockResolvedValueOnce(mockPapers as any);
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web/src/app/api/search/route.test.ts]: Fixed in 467a707: the fixtures are typed from the searchByKeyword return contract and contain valid Paper values without assertions.
